### PR TITLE
Update code to be compatible with Winterfell 0.8

### PR DESCRIPTION
**Pull request by Overcastan**
This PR updates the code to be compatible with new 0.8 version of the Winterfell.

This update mostly contains changes in `Felt` creation, since casting of `u64` to `FieldElement` was removed. It also updates the serialization and deserialization of the structs, which contain vectors of `Felt`s or `RpoDigest`s.

### CODE REVIEW

File: Cargo.toml
Comment (hackaugusto) [inline Cargo.toml:8]: maybe lets wait until this branch is merged downstream? Otherwise the changes LGTM

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**
